### PR TITLE
Send company profile data to API on submit

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-176)

will change this PR against master once [this](https://github.com/uktrade/directory-ui/pull/41) is merged.

once directory-api-client has stabilised we can install and use that client, but for now I have stubbed it with a mock. Will do a follow up PR to replace.

### CODE REVIEW

File: registration/templates/company-profile-update-error.html.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{% extends 'registration-base.html' %}
+
+{% block hero_title %}There was a problem updating your profile.{% endblock %}
+{% block hero_text %}Please try again.{% endblock %}
```
Comment (monikasulik) [inline registration/templates/company-profile-update-error.html.html]: Very unimportant, but nice to have a new line at the end of the file.

File: registration/views.py [hunk 2]
```diff
@@ -106,11 +106,19 @@ class CompanyProfileEditView(SessionWizardView):
...
-    def done(self, form_list, form_dict):
-        return redirect('company-detail')
+    def done(self, *args, **kwargs):
+
+        data = forms.serialize_company_profile_forms(self.get_all_cleaned_data())
+        response = api_client.company.update_profile(data)
+        if response.status_code == http.client.OK:
+            response = redirect('company-detail')
+        else:
+            response = TemplateResponse(self.request, self.failure_template)
```
Comment (monikasulik) [inline registration/views.py]: I suggest we add some error or warning logging here of what the response actually was.

**Comment by monikasulik** [comment]
Ah, hang on - just noticed. You probably want this to be against master not ED-140-company-profile-logo-field?

File: registration/views.py (hunk 2 above)
Comment (richtier) [inline registration/views.py]: good idea, lets definitely do that once the API client has been stabilised and we know the `response` structure. We should add the useful logging as a follow up PR. The reason I'm reluctant to do it yet is I'm unsure where the errors will be: will the errors be in `response.errors`,` response.content`, `response.json()`, `response.body`, for example

**Comment by richtier** [comment]
@monikasulik this branch relies on code in ` ED-140-company-profile-logo-field`. Once ` ED-140-company-profile-logo-field` has been merged I will amend this PR to be against master :)

File: registration/views.py (hunk 2 above)
Comment (monikasulik) [inline registration/views.py]: Fair enough :) As I said not crucial, but would be nice to have a ticket for it maybe in the backlog so it's not forgotten.

**Pull request merged**